stdlib/lang-parser/README.md

```diff
@@ -0,0 +1,309 @@
+## Ballerina-lang-parser
+
+The parser is implemented following the LL(K) recursive Descent parsing pattern. This pattern analyses the syntactic structure using k>1 lookahead tokens.
+
+**Token Structure**
+```ballerina
+Token record {
+    int tokenType;
+    string text;
+    int startPos;
+    int endPos;
+    int lineNumber;
+    int index;
+    string? whiteSpace;
+}
+```
+
+Grammar supported by the current implementation is given below.
+
+**Parser Grammar**
+```antlrv4
+Definition
+    | Function definition
+Function definition
+    | FUNCTION <callable Unit Signature> <callable Unit body>
+Callable Unit Signature
+    | IDENTIFIER ()
+Callable Unit Body
+    | { <statement*>}
+Statement
+    |<variable definition statement>
+    |<continue statement>
+Variable definition statement
+    | <valueTypeName> IDENTIFIER SEMICOLON
+    | <valueTypeName>  IDENTIFIER ASSIGN <expression> SEMICOLON
+Continue statement
+    | CONTINUE SEMICOLON
+valueTypeName
+    | INT
+Expression
+    | <simple literal>
+    | <variable reference>
+    | expression (DIVISION | MULTIPLICATION | MOD) expression
+    | expression ( ADD | SUB ) expression
+    | expression (LT_EQUAL | GT_EQUAL | GT | LT)expression
+    | expression (EQUAL | NOT_EQUAL) expression
+    | expression (REF_EQUAL | REF_NOT_EQUAL) expression
+    | (ADD | SUB | NOT | BIT_COMPLEMENT | UNTAINT) expression
+    | <tuple literal>
+    
+Simple Literal
+    | IntegerLiteral //[0-9]
+    | EmptyTupleLiteral
+
+Variable reference
+    | IDENTIFIER
+
+EmptyTupleLiteral
+    | LBRACE RBRACE
+
+TupleLiteral
+     |LEFT_PARENTHESIS expression (COMMA expression)* RIGHT_PARENTHESIS
+```
+
+
+#### Testing the parser
+
+1. Navigate to the <PROJECT_ROOT_DIRECTORY>/inputFile.txt and add the source code that is needed to be tested.
+
+	Sample source code:
+	```ballerina
+	function foo(){
+    	int a = 2 + (3*5);
+    }
+	```
+
+2. Run the `ballerina run parserModule` command from the <PROJECT_ROOT_DIRECTORY>
```

Comment:
Instead of `parserModule`, shall we just use `parser`?

If we also do the change to accept the file name as an argument we can change this to

```suggestion
2. Run the following command from the <PROJECT_ROOT_DIRECTORY>
`$ ballerina run parser test_file.bal`
```